easybuild/easyblocks/f/freefem.py

```diff
@@ -0,0 +1,109 @@
+        then configure FreeFem++ with the builded PETSc."""
+
+        # first Autoreconf has to be run
+        if not get_software_root('Autoconf'):
+            raise EasyBuildError("Autoconf is required to build FreeFem++. Please add it as build dependency")
+
+        run_cmd("autoreconf -i", log_all=True, simple=False)
+
+        # delete old installation, then set keeppreviousinstall to True (do not delete PETsc install)
+        self.make_installdir()
+        self.cfg['keeppreviousinstall'] = True
+
+        # configure and make petsc-slepc
+        cmd = "./configure --prefix=%s &&" % self.installdir
+        cmd += "cd download/ff-petsc &&"
+        cmd += "make petsc-slepc &&"
+        cmd += "cd ../.."
+        run_cmd(cmd, log_all=True, simple=False)
+
+        # check HDF5 and GSL deps and add the corresponding configopts
+        hdf5_root = get_software_root('HDF5')
+        if hdf5_root:
+            self.cfg.update('configopts', '--with-hdf5=%s ' % os.path.join(hdf5_root, 'bin', 'h5pcc'))
+
+        gsl_root = get_software_root('GSL')
+        if gsl_root:
+            self.cfg.update('configopts', '--with-gsl-include=%s ' % os.path.join(gsl_root, 'include'))
+            self.cfg.update('configopts', '--with-gsl-ldflags=%s ' % os.path.join(gsl_root, 'lib'))
+
+        # We should download deps
+        self.cfg.update('configopts', '--enable-download ')
+
+        # PaStiX does not work parallel, so disable it.
+        self.cfg.update('configopts', '--disable-pastix ')
+
+        super(EB_FreeFem_plus__plus_, self).configure_step()
+
+    def build_step(self):
+        # dependencies are heavily patched by FreeFem++, we therefore let it download and install them itself
+        self.cfg.update('prebuildopts', 'download/getall -a -o ScaLAPACK,ARPACK,freeYams,Gmm++,Hips,Ipopt,METIS,'
+                        'ParMETIS,MMG3D,mshmet,MUMPS,NLopt,pARMS,PaStiX,Scotch,SuiteSparse,SuperLU_DIST,SuperLU,'
+                        'TetGen,PETSc,SLEPc,hpddm &&')
+
+        # FreeFem++ parallel make does not work.
+        if self.cfg['parallel'] != 1:
+            self.log.warning("Ignoring requested build parallelism, it breaks FreeFem++ building, so setting to 1")
+            self.cfg['parallel'] = 1
+
+        super(EB_FreeFem_plus__plus_, self).build_step()
+
+    def sanity_check_step(self):
+        # define FreeFem++ sanity_check_paths here
+
+        custom_paths = {
+            'files': ['bin/%s' % x for x in ['bamg', 'cvmsh2', 'ffglut', 'ffmedit']] +
+            ['bin/ff-%s' % x for x in ['c++', 'get-dep', 'mpirun', 'pkg-download']] +
+            ['bin/FreeFem++%s' % x for x in ['', '-mpi', '-nw']],
+            'dirs': ['share/freefem++/%s' % self.version] +
+            ['lib/ff++/%s/%s' % (self.version, x) for x in ['bin', 'etc', 'idp', 'include', 'lib']]
+        }
+        super(EB_FreeFem_plus__plus_, self).sanity_check_step(custom_paths=custom_paths)
```

Comment:
I think sanity check step should return the result. @boegel I that true in all cases or only when dealing with extensions?

Reply:
I do not understand this comment. I thought that I just hard-coded in the easyblock the sanity-check files and dirs, nothing else. (So we do not have to put in the easyconfig)